### PR TITLE
fix: Smartrate deserialization

### PR DESCRIPTION
**Pull request by nwithan8**
- adds new SmartrateCollection object to collect a list of Rate objects
- adds new SmartrateDeserializer to handle parsing smartrate-specific response JSON schema
- utilizes RateDeserializer to parse Rate objects, store in SmartrateCollection object
- getSmartrates() function utilizes SmartrateCollection internally, returns list of Rate objects (no change in return type)

I am not sure about testing procedures yet, so this is just a draft

### CODE REVIEW

File: src/main/java/com/easypost/model/Shipment.java [hunk 1]
```diff
@@ -291,7 +291,6 @@ public Shipment newRates(Map<String, Object> params, String apiKey) throws EasyP
 	}
 
 	// get smartrates
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/Shipment.java:293]: Can we unindent this?

**Review by Justintime50 — CHANGES_REQUESTED** [review]

**Review by Justintime50 — APPROVED** [review]
Code LGTM, tested with the team and it works properly.

This will merge into @jontsai's branch to complete it and is necessary to move forward with SmartRate.

**Review by Sarahbear1926 — APPROVED** [review]

**Comment by nwithan8** [comment]
Re-added some things that got lost in the b88fe51 rollback.
Tests run correctly locally.

**Review by ajnassar — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]
LGTM, tested it with the team and the code works as we expected.

**Pull request merged**